pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -171,5 +181,97 @@ func (a *AgentClusterInstall) finish() error {
 		return errors.New("missing configuration or manifest file")
 	}
 
+	if err := a.validateIPAddressAndNetworkType().ToAggregate(); err != nil {
+		return errors.Wrapf(err, "invalid NetworkType configured")
+	}
+
 	return nil
 }
+
+// Sets the default network type to OVNKubernetes if it is unspecified in the
+// AgentClusterInstall or InstallConfig
+func setNetworkType(aci *hiveext.AgentClusterInstall, installConfig *types.InstallConfig,
+	warningMessage string) {
+
+	if aci.Spec.Networking.NetworkType != "" {
+		return
+	}
+
+	if installConfig != nil && installConfig.Networking != nil &&
+		installConfig.Networking.NetworkType != "" {
+		aci.Spec.Networking.NetworkType = installConfig.NetworkType
+		return
+	}
+
+	defaults.SetInstallConfigDefaults(installConfig)
+	logrus.Infof("%s Defaulting NetworkType to %s.", warningMessage, installConfig.NetworkType)
+	aci.Spec.Networking.NetworkType = installConfig.NetworkType
+}
+
+func isIPv4(ipAddress net.IP) bool {
+	ip := ipAddress.To4()
+	if ip == nil {
+		return false
+	}
+	return true
+}
+
+func isIPv6(ipAddress net.IP) bool {
+	ip := ipAddress.To16()
+	if ip == nil {
+		return false
+	}
+	return true
+}
+
+func (a *AgentClusterInstall) validateIPAddressAndNetworkType() field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	fieldPath := field.NewPath("spec", "networking", "networkType")
+	clusterNetworkPath := field.NewPath("spec", "networking", "clusterNetwork")
+	serviceNetworkPath := field.NewPath("spec", "networking", "serviceNetwork")
+
+	if a.Config.Spec.Networking.NetworkType == string(operv1.NetworkTypeOpenShiftSDN) {
+		hasIPv6 := false
+		hasIPv4 := false
+		for _, cn := range a.Config.Spec.Networking.ClusterNetwork {
+			ip, _, errCIDR := net.ParseCIDR(cn.CIDR)
+			if errCIDR != nil {
+				allErrs = append(allErrs, field.Required(clusterNetworkPath, "error parsing the clusterNetwork CIDR"))
+			}
+			if isIPv4(ip) {
+				hasIPv4 = true
+			}
+			if isIPv6(ip) {
+				hasIPv6 = true
+			}
+		}
+		if hasIPv6 && !hasIPv4 {
```

Comment:
Is OVN not required for dual stack?  I don't really know for sure but I can check.

Reply:
The validation returns an error if the network type is OpenShiftSDN and an IPv6 address has been entered. This should catch the dual stack case. The validation will fail if they selected OpenShiftSDN and would lead the user to change to a different network type.